openff/bespokefit/executor/services/qcgenerator/cache.py

```diff
@@ -70,16 +162,13 @@ def cached_compute_task(
     # Canonicalize the task to improve the cache hit rate.
     task = _canonicalize_task(task)
```

Comment:
I think we need to do this first to maximise the cache hit rate, this will get skipped if the task is a `Torsion1DTask` as we return the method straight away.

Reply:
I think to work with my other comment above we could just reorder the function, I would do the canonicalize and cache look up first then the rest of the function as is after.
```suggestion
    task = _canonicalize_task(task)
    task_hash = _sha256_hash(task.json())
    task_id = _retrieve_cached_task_id(task_hash, redis_connection)
    
    if task_id is not None:
        return task_id
       
    if isinstance(task, Torsion1DTask):
        compute = worker.compute_torsion_drive
        return _compute_torsion_drive_task(task, redis_connection)
    elif isinstance(task, OptimizationTask):
        compute = worker.compute_optimization
    elif isinstance(task, HessianTask):
        compute = worker.compute_hessian
    else:
        raise NotImplementedError()
        
    task_id = compute.delay(task_json=task.json()).id
    _cache_task_id(task_id, task.type, task_hash, redis_connection)
    return task_id
```

Reply:
> I think we need to do this first to maximise the cache hit rate, this will get skipped if the task is a Torsion1DTask as we return the method straight away.

Ah good catch - I missed this after shuffling the logic around!